core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -256,6 +256,7 @@ class Partition(val topicPartition: TopicPartition,
   // start offset for 'leaderEpoch' above (leader epoch of the current leader for this partition),
   // defined when this broker is leader for partition
   @volatile private var leaderEpochStartOffsetOpt: Option[Long] = None
+  // Point to itself if is leader, otherwise point to the latest leader replica
```

Comment:
nit: I would phrase it as follow: `Replica ID of the leader, defined when this broker is leader or follower for the partition.`.